apps/meteor/app/livechat/server/lib/Livechat.js

```diff
@@ -678,24 +677,6 @@ export const Livechat = {
 		return updateDepartmentAgents(_id, departmentAgents, department.enabled);
 	},
 
-	async saveAgentInfo(_id, agentData, agentDepartments) {
-		check(_id, Match.Maybe(String));
-		check(agentData, Object);
-		check(agentDepartments, [String]);
-
-		const user = await Users.findOneById(_id);
-		if (!user || !(await hasRoleAsync(_id, 'livechat-agent'))) {
-			throw new Meteor.Error('error-user-is-not-agent', 'User is not a livechat agent', {
-				method: 'livechat:saveAgentInfo',
-			});
-		}
-
-		await Users.setLivechatData(_id, agentData);
-		await LivechatDepartmentRaw.saveDepartmentsByAgent(user, agentDepartments);
-
-		return true;
-	},
-
```

Comment:
Kinda. The fix was to replicate this function in other place, with some tweaks.

Since we were changing the function behavior and we had to use it in other place too, we moved it to the .ts file.

Not ideal, but since the func was smol, also don't think it's a big deal _for this specific case_

Reply:
Same situation with the function that was inside the model. We had to reuse it, so we moved it outside of the model to do that.

That way the model keeps clean too